addons/hw_drivers/controllers/driver.py

```diff
@@ -1,432 +1,325 @@
-            subprocess.call("fswebcam -d " + adrress + " /tmp/testimage -r " + picture, shell=True)
-            image_bytes = subprocess.check_output('cat /tmp/testimage | base64',shell=True)
-            result = {'image': image_bytes}
-        return result
-
-    @http.route('/hw_drivers/send_iot_box', type='http', auth='none', cors='*')
-    def send_iot_box(self):
-        send_iot_box_device(False)
-        return 'ok'
 
 #----------------------------------------------------------
-# Driver common interface
+# Helper
 #----------------------------------------------------------
-class Driver(Thread):
-    ping_value = ""
 
-    def supported(self):
-        pass
+def get_mac_address():
+    try:
+        return netifaces.ifaddresses('eth0')[netifaces.AF_LINK][0]['addr']
+    except:
+        return netifaces.ifaddresses('wlan0')[netifaces.AF_LINK][0]['addr']
 
-    def value(self):
-        pass
+def get_ip():
+    try:
+        return netifaces.ifaddresses('eth0')[netifaces.AF_INET][0]['addr']
+    except:
+        return netifaces.ifaddresses('wlan0')[netifaces.AF_INET][0]['addr']
 
-    def get_name(self):
-        pass
+def read_file_first_line(filename):
+    content = ""
+    try:
+        path = os.getenv("HOME") + '/' + filename
+        f = open(path, 'r')
+        content = f.readline().strip('\n')
+        f.close()
+    finally:
+        return content
 
-    def get_connection(self):
-        pass
+def get_odoo_server_url():
+    return read_file_first_line('odoo-remote-server.conf')
 
-    def action(self, action):
-        pass
+def get_token():
+    return read_file_first_line('token')
 
 #----------------------------------------------------------
-# Usb drivers
+# Controllers
 #----------------------------------------------------------
-usbdrivers = []
-drivers = {}
-
-class UsbMetaClass(type):
-    def __new__(cls, clsname, bases, attrs):
-        newclass = super(UsbMetaClass, cls).__new__(cls, clsname, bases, attrs)
-        usbdrivers.append(newclass)
-        return newclass
-
-class USBDriver(Driver,metaclass=UsbMetaClass):
-    def __init__(self, dev):
-        super(USBDriver, self).__init__()
-        self.dev = dev
-        self.value = ""
-
-    def get_name(self):
-        lsusb = str(subprocess.check_output('lsusb')).split("\\n")
-        for usbpath in lsusb:  # Should filter on usb devices or inverse loops?
-            device = self.dev
-            if "%04x:%04x" % (device.idVendor, device.idProduct) in usbpath:
-                return usbpath.split("%04x:%04x" % (device.idVendor, device.idProduct))[1]
-        return str(device.idVendor) + ":" + str(device.idProduct)
-
-    def get_connection(self):
-        return 'direct'
-
-    def value(self):
-        return self.value
 
+class StatusController(http.Controller):
+    @http.route('/hw_drivers/action', type='json', auth='none', cors='*', csrf=False)
+    def action(self, device_id, data):
+        if device_id in iot_devices:
```

Comment:
```
iot_device = iot_devices.get('device_id')
if iot_device
    return iot_device.action(data)
return False
```

does the webclient really wants to know that the device id was not found? i'm not sure

Reply:
i would make the request fail + add a readable message